CenterForOpenScience/osf.io
# [feature] Add Preprint Filter to OSF Search [OSF-6978]

#### Purpose
- Adds a "Preprints" filter to OSF search results.
  - When clicked, the filter redirects to the preprint search page with the current query and OSF provider filter applied (i.e. https://osf.io/preprints/discover/?provider=OSF&q=science).
- If a project that appears in the search results has an associated preprint, there will be a "Jump to Preprint" link that links to the preprint detail page (i.e. osf.io/<preprint_guid>/), per discussion with Nici.

#### Changes
- Adds the `preprint` doc_type to elasticsearch.
- Adds the `preprint` template to search.mako.
- Adds the `preprint` filter (with appropriate link) to the search page. 

#### Notes
- `migrate_search` needs to be run.

#### QA Notes
- Check the search page on the OSF, but also check the API search endpoint.
  - `/v2/search/preprints/` doesn't exist as of this PR, but `/v2/search/` and all of the extended search endpoints should still work as expected.

#### Side effects
- None expected.

#### Ticket
- [OSF-6978](https://openscience.atlassian.net/browse/OSF-6978)

#### Screenshots 
- Search Filters:
  - "Preprint" links to the preprint search page.
![screen shot 2016-11-04 at 4 56 08 pm](https://cloud.githubusercontent.com/assets/7913604/20022181/d07e0af8-a2af-11e6-8d42-27757372e18d.png)

- Project with Preprint in Search Results (see first result):
  - "Jump to Preprint" links to the detail page of the *preprint*, while the project title links to the detail page of the *project*.
![screen shot 2016-11-04 at 4 56 45 pm](https://cloud.githubusercontent.com/assets/7913604/20022198/de789bdc-a2af-11e6-8101-6b695dedce86.png)
